src/librustc_ast_pretty/pprust.rs

```diff
@@ -102,21 +97,22 @@ crate const INDENT_UNIT: usize = 4;
 /// it can scan the input text for comments to copy forward.
 pub fn print_crate<'a>(
     cm: &'a SourceMap,
-    sess: &ParseSess,
     krate: &ast::Crate,
     filename: FileName,
     input: String,
     ann: &'a dyn PpAnn,
     is_expanded: bool,
+    edition: Edition,
+    injected_crate_name: &Once<Symbol>,
```

Comment:
```suggestion
    has_injected_crate: bool,
```
`Once` and even `Option` are not needed here.